enhancements/cluster-logging/cluster-logging-move-init-to-es-operator.md

```diff
@@ -0,0 +1,80 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@ewolinetz"
+reviewers:
+  - "@bparees"
+  - "@jcantrill"
+  - "@igor-karpukhin"
+  - "@alanconway"
+  - "@ewolinetz"
+approvers:
+  - "@ewolinetz"
+  - "@jcantrill"
+creation-date: 2019-09-17
+last-updated: 2019-10-25
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-es-init
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The purpose of moving Elasticsearch initialization logic to the elasticsearch-operator is to reduce the number of times these init scripts executed.
+At the moment we keep these scripts in the ES image, and execute them everytime when ES pod starts. 
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that allows users to specify init procedures for the ES cluster
+* Allow users to choose what command has to be executed during the init process for NEW and UPGRADE elasticsearch cluster states
+
+## Proposal
+
+The following changes will define how the elasticsearch-operator executes init process for Elasticsearch cluster fresh start and upgrade
+
+* Scripts will be executed from outside the ES image
+* Adding additional section to the elasticsearch-operator called `actions`, which may look like this:
+
+    ```json
+    "actions": {
```

Comment:
@ewolinetz @jcantrill I just thought about multiple entries for the `action` section. Would it be good to have more than one? What if we would need to execute multiple commands from multiple images. Like for example, the first Job executes initialization for Jaeger, and then the second one will run another job for seeding index templates and do other stuff. That would give us a bit more flexibility, not to put all possible initialization logic into one image.

Reply:
I think keeping it to a single image and a single script that can handle this logic allows users to accomplish initialization and keeps it simple for our operator. If the cluster requires different things done, I would argue it should all be contained in the same place.

Reply:
I think its reasonable to require a single image that wraps up all of your initialization requirements so a single image field is sufficient